src/yarp.c

```diff
@@ -2621,6 +2621,12 @@ parse_expression_prefix(yp_parser_t *parser) {
       return yp_node_class_variable_read_create(parser, &parser->previous);
     case YP_TOKEN_CONSTANT:
       return yp_node_constant_read_create(parser, &parser->previous);
+    case YP_TOKEN_COLON_COLON: {
+      yp_token_t delimiter = parser->previous;
+      yp_node_t *child = parse_expression(parser, BINDING_POWER_CALL, "Expected constant after ::");
```

Comment:
I think Ruby doesn't allow anything other than a constant after `::`, and if I'm not mistaken `parse_expression` will accept something like `::a` here.

Reply:
No worries! @kddnewton could you please take a look at both implementations and decide if we want a new node or keep current ones? I'll fix my implementation if we decide to go with this direction because yeah, it allows doing `::A::b` now 😞